### PR TITLE
Store Transaction Overview and Pagination

### PR DESCRIPTION
**Pull request by Naijabas**
# Description

Please include a summary of the change and which issue is fixed. Add a refrence to the issue that was fixed! 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnin

### CODE REVIEW

File: app/Http/Controllers/StoreController.php [hunk 1]
```diff
@@ -225,7 +225,7 @@ public function show(Request $request, $id)
 
             if ($statusCode == 200  && $transaction_statusCode == 200) {
 
-                return view('backend.stores.show')->with('response', $StoreData);
+                return view('backend.stores.show')->with('response', $StoreData)->with('number', 1);
```
Comment (Jeremiahiro) [inline app/Http/Controllers/StoreController.php:228]: why return with number?

File: resources/views/backend/stores/show.blade.php [hunk 2]
```diff
@@ -184,38 +188,59 @@ class="ml-2 text-truncate">From previous Month</span>
...
+                                            <th>Customer Name </th>
+                                            <th data-priority="1">Amount</th>
+                                            <th data-priority="3">Transaction Type</th>
+                                            <th data-priority="1">Status</th>
+                                            <th data-priority="3"> </th>
+                                        </tr>
+                                        </thead>
+                                        <tbody>
+                                        @foreach ($response['storeData']->customers as $transactions)
+
```
Comment (Jeremiahiro) [inline resources/views/backend/stores/show.blade.php:209]: you could have done this in the controller

$transactions = $response->storeData->customers;

File: resources/views/backend/stores/show.blade.php [hunk 3]
```diff
@@ -184,38 +188,59 @@ class="ml-2 text-truncate">From previous Month</span>
...
+                                        
+                                        <tr>
+                                            <td>{{ $number++ }}</td>
+                                            <th>{{$transactions->name}}<span class="co-name"></span>
+                                                <br> <span class="font-light">{{$transactions->phone_number}}</span>
+                                            </th>
+                                            <td>{{$transaction->amount}}</td>
+                                            <td>{{$transaction->type}}</td>
+                                            <td>
+                                            @if ($transaction->status == 0)
```
Comment (Jeremiahiro) [inline resources/views/backend/stores/show.blade.php:220]: the unpaid and paid status does not have a style

File: resources/views/backend/stores/show.blade.php [hunk 4]
```diff
@@ -184,38 +188,59 @@ class="ml-2 text-truncate">From previous Month</span>
...
+                            <div class="card">
+                            <h4 class="card-title">{{ ucfirst($storeData->store_name) }} Transaction Overview</h4>
+                                    
+                                    <table id="datatable-buttons" class="table dt-responsive">
+                                        <thead>
+                                        <tr>
+                                            <th>S/N</th>
+                                            <th>Customer Name </th>
+                                            <th data-priority="1">Amount</th>
+                                            <th data-priority="3">Transaction Type</th>
```
Comment (Jeremiahiro) [inline resources/views/backend/stores/show.blade.php:202]: what function does the data-priority play?

File: resources/views/backend/stores/show.blade.php [hunk 5]
```diff
@@ -227,6 +252,20 @@ class="btn btn-primary waves-effect waves-light"> View Transaction</a>
 
 @section("javascript")
 <script src="/backend/assets/build/js/intlTelInput.js"></script>
+<script src="/backend/assets/libs/datatables/jquery.dataTables.min.js"></script>
```
Comment (Jeremiahiro) [inline resources/views/backend/stores/show.blade.php:255]: you could move all of this to the layout.app

File: resources/views/backend/transaction/show.blade.php [hunk 6]
```diff
@@ -134,7 +134,7 @@ class="form-control w-100 flex-1">{{ $transaction->description }}</textarea>
                                                             <tr class="font-weight-bolder">
                                                                 <th scope="row">Total Amount</th>
                                                                 <td colspan="2">
-                                                                    {{ floor($transaction->amount/100+12 * $transaction->interest + $transaction->amount) }}
+                                                                    {{ (($transaction->interest / 100) * $transaction->amount) + $transaction->amount }}
```
Comment (Jeremiahiro) [inline resources/views/backend/transaction/show.blade.php:137]: the floor is necessary as it removes decimals

Comment (Jeremiahiro) [reply resources/views/backend/transaction/show.blade.php:137]: the calculation you have here, is it for monthly or manual?

File: app/Http/Controllers/StoreController.php (hunk 1 above)
Comment (Naijabas) [reply app/Http/Controllers/StoreController.php:228]: I'm using it to make the s/n

File: resources/views/backend/stores/show.blade.php (hunk 2 above)
Comment (Naijabas) [reply resources/views/backend/stores/show.blade.php:209]: That's not the only place using the data

I don't wanna cause problem for somebody working on it

Is this a problem?

File: resources/views/backend/transaction/show.blade.php (hunk 6 above)
Comment (Naijabas) [reply resources/views/backend/transaction/show.blade.php:137]: I didn't do it

It was after I merged and it caused conflict so I have to commit it with my files

File: resources/views/backend/stores/show.blade.php (hunk 4 above)
Comment (Naijabas) [reply resources/views/backend/stores/show.blade.php:202]: It's just there

Comment (Naijabas) [reply resources/views/backend/stores/show.blade.php:202]: I met it there tho

File: resources/views/backend/stores/show.blade.php (hunk 3 above)
Comment (Naijabas) [reply resources/views/backend/stores/show.blade.php:220]: A style?

What's that?

File: app/Http/Controllers/StoreController.php (hunk 1 above)
Comment (Naijabas) [reply app/Http/Controllers/StoreController.php:228]: To loop through it

File: resources/views/backend/stores/show.blade.php (hunk 5 above)
Comment (Naijabas) [reply resources/views/backend/stores/show.blade.php:255]: Or layout.base?

File: app/Http/Controllers/StoreController.php (hunk 1 above)
Comment (Jeremiahiro) [reply app/Http/Controllers/StoreController.php:228]: there are many ways you could use to do this, this is so its easy for other developers to understand the code

File: resources/views/backend/stores/show.blade.php (hunk 2 above)
Comment (Jeremiahiro) [reply resources/views/backend/stores/show.blade.php:209]: ok, that's fine

File: resources/views/backend/stores/show.blade.php (hunk 4 above)
Comment (Jeremiahiro) [reply resources/views/backend/stores/show.blade.php:202]: please remove it and see what happens

File: resources/views/backend/stores/show.blade.php (hunk 5 above)
Comment (Jeremiahiro) [reply resources/views/backend/stores/show.blade.php:255]: yes layout.base

File: resources/views/backend/transaction/show.blade.php (hunk 6 above)
Comment (Jeremiahiro) [reply resources/views/backend/transaction/show.blade.php:137]: ok

**Pull request merged**